src/core/shuffler.js

```diff
@@ -0,0 +1,28 @@
+import { deepCopyGrid, listCoordsMovableTiles, move } from './game';
+
+export const SHUFFLE_DURATION = 1000;
+export const SLEEP_DURATION = 1;
+
+export const choiceCoords = coordsList =>
+    coordsList[Math.floor(Math.random() * coordsList.length)];
+
+const sleep = duration => {
+    return new Promise(resolve => setTimeout(resolve, duration));
+};
+
+export const shuffle = async grid => {
+    let shuffledGrid = deepCopyGrid(grid);
+
+    const movePromise = new Promise(() => {
+        let continueShuffling = true;
+        while (continueShuffling) {
```

Comment:
Do you set `continueShuffling` to `false` anywhere ?

Reply:
Eslint: "unexpected constant condition"

I'll use:
// eslint-disable-next-line no-constant-condition